### PR TITLE
Create open_redirect_pirlsandiego.yml

### PR DESCRIPTION
**Pull request by zoomequipd**
# Description
Match message containing an open redirect from PIRL San Diego
# Associated samples

- [Sample 1](https://platform.sublimesecurity.com/messages/ce5e11d10c4b26493f4e00c851dce757b40ca047470924b033a88b721361933e)

## Associated hunts

- [Hunt 1](https://platform.sublimesecurity.com/hunts/e33f67e7-0877-447a-ace8-4b5d887b6ee7)

### CODE REVIEW

**Comment by zoomequipd** [comment]
Low volume hits while deployed in test.  Ready for review

File: detection-rules/open_redirect_pirlsandiego.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  Message contains use of the PIRL San Deigo open redirect. This has been exploited in the wild.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.root_domain == "pirlsandiego.net"
+          and strings.icontains(.href_url.path, '/LinkClick.aspx')
+          and strings.icontains(.href_url.url, 'link=')
+  )
```
Comment (aidenmitchell) [inline detection-rules/open_redirect_pirlsandiego.yml:12]:
```suggestion
  )
  and sender.email.domain.root_domain != 'pirlsandiego.net'
```

Comment (zoomequipd) [reply detection-rules/open_redirect_pirlsandiego.yml:12]: `pirlsandiego.net` doesn't have an MX record, so they must not be sending email from that.

Based on what I could find on their website, this is used to redirect users out of pirlsandiego.net, lots of examples on this page: http://pirlsandiego.net/Npirl/ContactInfo/tabid/541/Default.aspx

I'm confident it'll be abnormal to see this redirect in messages.

If we want to add confidence, we might be able to negate links where `tabid` and `mid` are present within the URL query params, which hasn't been observed in malicious use but is present on the website.  If we start seeing FPs, I'll get that added in quickly.

Comment (aidenmitchell) [reply detection-rules/open_redirect_pirlsandiego.yml:12]: Makes sense, thanks for the context 🫡

**Review by aidenmitchell — APPROVED** [review]

**Pull request merged**